langchain/src/document_loaders/web/wikipedia.ts

```diff
@@ -0,0 +1,131 @@
+import wiki, { Page } from "wikipedia";
+import { Document } from "../../document.js";
+import { BaseDocumentLoader } from "../base.js";
+
+const DISCARDCATEGORIES = [
+  "See also",
+  "References",
+  "External links",
+  "Further reading",
+  "Footnotes",
+  "Bibliography",
+  "Sources",
+  "Citations",
+  "Literature",
+  "Footnotes",
+  "Notes and references",
+  "Photo gallery",
+  "Works cited",
+  "Photos",
+  "Gallery",
+  "Notes",
+  "References and sources",
+  "References and notes",
+];
+
+export class WikipediaLoader extends BaseDocumentLoader {
+  discardCategories: string[];
+
+  constructor(public webPath: string, discardCategories?: string[]) {
+    super();
+    this.discardCategories = discardCategories ?? DISCARDCATEGORIES;
+  }
+
+  public async load(): Promise<Document[]> {
+    const matchResult = this.webPath.match(/\/wiki\/(.+)/);
+    if (!matchResult) {
+      throw new Error(
+        "Invalid webPath format. Use full url like https://en.wikipedia.org/wiki/2020_Summer_Olympics"
+      );
+    }
+
+    const pageTitle = decodeURIComponent(matchResult[1]);
+    const page: Page = await wiki.page(pageTitle);
+    const content = await page.content();
+    const sections = this.extractSections(
+      content,
+      page.title,
+      this.discardCategories
+    );
+
+    const docs = sections.map(
+      (section) =>
+        new Document({
+          pageContent: section,
+          metadata: { source: this.webPath },
+        })
+    );
+    return docs;
+  }
+
+  // converted from the openai cookbook example
+  // https://github.com/openai/openai-cookbook/blob/main/examples/fine-tuned_qa/olympics-1-collect-data.ipynb
+  private extractSections(
```

Comment:
Would rather see section splitting being done by a splitter instead of by the loader.